### PR TITLE
Using a single executor for buffered mutator per connection.

### PR DESCRIPTION
**Pull request by sduskis**
This should be the last PR for #630.

### CODE REVIEW

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java [hunk 1]
```diff
@@ -91,7 +90,7 @@ public void run() {
     }
   }
 
-  private static ExecutorService createExecutorService(BigtableOptions options) {
+  public static ExecutorService createExecutorService(BigtableOptions options) {
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: createBufferedMutatorExecutorService
and add some doc

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java [hunk 2]
```diff
@@ -178,18 +178,18 @@ public BigtableBufferedMutator(
       BufferedMutator.ExceptionListener listener,
       HeapSizeManager heapSizeManager) {
     this(client, adapter, configuration, options, listener, heapSizeManager,
-        createExecutorService(options));
+        createExecutorService(options), true);
   }
 
-  @VisibleForTesting
-  BigtableBufferedMutator(
+  public BigtableBufferedMutator(
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: Is this constructor for testing only? if yes, annotate it with @VisibleForTesting

Comment (sduskis) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: We're actually only using this constructor.  I'll remove the other one.

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java (hunk 1 above)
Comment (sduskis) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: I'll move this to AbstractBigtableConnection and the test.

File: bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java [hunk 3]
```diff
@@ -184,6 +185,10 @@ public BufferedMutator getBufferedMutator(BufferedMutatorParams params) throws I
 
     final long id = SEQUENCE_GENERATOR.incrementAndGet();
 
+    if (this.bufferedMutatorExecutorService == null) {
+      this.bufferedMutatorExecutorService = BigtableBufferedMutator.createExecutorService(options);
```
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java]: this is created, but never used (except .shutdown())?

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java (hunk 1 above)
Comment (sduskis) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: Done

Comment (sduskis) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: I'll add docs here.  Isn't BigtableBufferedMutator. createBufferedMutatorExecutorService() redundant?

File: bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java (hunk 3 above)
Comment (sduskis) [inline bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/AbstractBigtableConnection.java]: I fixed that.

File: bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java (hunk 1 above)
Comment (kevinsi4508) [inline bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java]: You are right, "ExecutorService" alone is kind of generic. "WorkerExecutorService"?

**Comment by kevinsi4508** [comment]
LGTM

**Pull request merged**